pkg/serverless/invocationlifecycle/trace.go

```diff
@@ -39,17 +41,17 @@ var currentExecutionInfo executionStartInfo
 
 // startExecutionSpan records information from the start of the invocation.
 // It should be called at the start of the invocation.
-func startExecutionSpan(startTime time.Time, rawPayload string) {
+func startExecutionSpan(startTime time.Time, rawPayload string, isInferredSpan bool) {
 	currentExecutionInfo.startTime = startTime
-	currentExecutionInfo.traceID = random.Uint64()
-	currentExecutionInfo.spanID = random.Uint64()
+	currentExecutionInfo.traceID = rand.Random.Uint64()
+	currentExecutionInfo.spanID = rand.Random.Uint64()
 	currentExecutionInfo.parentID = 0
 
 	payload := convertRawPayload(rawPayload)
 
-	if payload.Headers != nil {
+	if payload.Headers != nil && !isInferredSpan {
```

Comment:
I removed this logic as it was unnecessary. We can just overwrite them in the endExecutionSpan function.